peterjokumsen/peterjokumsen-nx-workspace
# Update workflow only execture infra jobs when `.bicep` files have changed

## Summary

Check if there are `.bicep` file changes between current `HEAD` and `BASE`, getting base SHA from `nrwl/nx-set-shas@v4` action.

## Changes Made

- Renamed primary workflow to `CI/CD`, to be extended for any other applications created in the mono-repo. 
- Moved affected check for application to `preparation` workflow.
- Updated affected check to also check if `.bicep` files have been changed, output as `infra_affected`
- Updated `infra` workflow to take in `infra_affected` property and only run lint/deploy jobs when value is `'Yes'`
- Build job updated to build application when `infra_affected` is `'Yes'` to make sure that application is deployed when infra is deployed.

## Checklist

- [x] I have added comments to code in hard-to-understand areas

<details>
<summary><strong>Expand for optional sections</strong></summary>

## Related issues

Resolves #78

## Testing instructions

Will be applying tests in this PR, will push a change to `.bicep` files and confirm whether infra jobs are run.

### First run without `.bicep` file changes
![image](https://github.com/peterjokumsen/peterjokumsen-nx-workspace/assets/7055026/2a6e5761-5f19-48ae-b47b-2c6dee878e7b)
> Lint and deploy jobs skipped

### Run with `.bicep` change made
![image](https://github.com/peterjokumsen/peterjokumsen-nx-workspace/assets/7055026/25bfd013-2cee-4aa3-9378-65bca9f25d89)
> Build application, and Lint & Deploy infra jobs run when change made to `main.bicep`

### Run with change made to application only
![image](https://github.com/peterjokumsen/peterjokumsen-nx-workspace/assets/7055026/894a3bbc-6236-48ee-9ee5-7a4838c75be5)
> Lint & Deploy infra jobs skipped, build and test application jobs done as expected.

</details>
